File: algorithms/c/70.c
#include <stdio.h>

int climbStairs(int n)
{
    if ( n == 0 )
        return 0;
    int p, q, t;
    p = q = 1;
    for ( int i = 1; i < n; i++ )
    {
        t = p+q;
        p = q;
        q = t;
    }
    return q;
}

int main()
{
    printf("%d\n", climbStairs(3));
    return 0;
}

File: algorithms/python/70.py
from functools import lru_cache

class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        @lru_cache(maxsize = None)
        def helper(n):
            if n == 1:
                return 1
            elif n == 2:
                return 2
            else:
                return helper(n-2) + helper(n-1)

        return helper(n)
